tests/contrib/molten/test_molten.py

```diff
@@ -0,0 +1,215 @@
+        response = molten_client()
+        spans = self.tracer.writer.pop()
+        self.assertEqual(response.status_code, 200)
+        # TestResponse from TestClient is wrapper around Response so we must
+        # access data property
+        self.assertEqual(response.data, '"Hello 24 year old named Jim!"')
+        span = spans[0]
+        self.assertEqual(span.service, 'molten')
+        self.assertEqual(span.name, 'molten.request')
+        self.assertEqual(span.resource, 'GET /hello/{name}/{age}')
+        self.assertEqual(span.get_tag('http.method'), 'GET')
+        self.assertEqual(span.get_tag('http.url'), '/hello/Jim/24')
+        self.assertEqual(span.get_tag('http.status_code'), '200')
+
+        # See test_resources below for specifics of this difference
+        if MOLTEN_VERSION >= (0, 7, 2):
+            self.assertEqual(len(spans), 18)
+        else:
+            self.assertEqual(len(spans), 16)
+
+        # test override of service name
+        Pin.override(molten, service=self.TEST_SERVICE)
+        response = molten_client()
+        spans = self.tracer.writer.pop()
+        self.assertEqual(spans[0].service, 'molten-patch')
+
+    def test_route_failure(self):
+        app = molten.App(routes=[molten.Route('/hello/{name}/{age}', hello)])
+        client = TestClient(app)
+        response = client.get('/goodbye')
+        spans = self.tracer.writer.pop()
+        self.assertEqual(response.status_code, 404)
+        span = spans[0]
+        self.assertEqual(span.service, 'molten')
+        self.assertEqual(span.name, 'molten.request')
+        self.assertEqual(span.resource, 'GET 404')
+        self.assertEqual(span.get_tag('http.url'), '/goodbye')
+        self.assertEqual(span.get_tag('http.method'), 'GET')
+        self.assertEqual(span.get_tag('http.status_code'), '404')
+
+    def test_route_exception(self):
+        def route_error() -> str:
+            raise Exception('Error message')
+        app = molten.App(routes=[molten.Route('/error', route_error)])
+        client = TestClient(app)
+        response = client.get('/error')
+        spans = self.tracer.writer.pop()
+        self.assertEqual(response.status_code, 500)
+        span = spans[0]
+        route_error_span = spans[-1]
+        self.assertEqual(span.service, 'molten')
+        self.assertEqual(span.name, 'molten.request')
+        self.assertEqual(span.resource, 'GET /error')
+        self.assertEqual(span.error, 1)
+        # error tags only set for route function span and not root span
+        self.assertIsNone(span.get_tag(errors.ERROR_MSG))
+        self.assertEqual(route_error_span.get_tag(errors.ERROR_MSG), 'Error message')
+
+    def test_resources(self):
+        """ Tests request has expected span resources """
+        response = molten_client()
+        spans = self.tracer.writer.pop()
+
+        # `can_handle_parameter` appears twice since two parameters are in request
+        # TODO[tahir]: missing ``resolve` method for components
```

Comment:
outdated comment?

Reply:
Maybe `TODO` is stronger than I intended it. We want to revisit this limitation of the current instrumentation and we have it on our backlog.